gtdynamics/factors/ContactPointFactor.h

```diff
@@ -112,4 +112,96 @@ class ContactPointFactor
   }
 };
 
+/**
+ * ContactPoseFactor is a two-way nonlinear factor which constrains a link pose
+ * and a reference frame defined at a point of contact P in the world/spatial
+ * frame.
+ *
+ * This factor is useful for implementing flat foot constraints.
+ */
+class ContactPoseFactor
+    : public gtsam::NoiseModelFactor2<gtsam::Pose3, gtsam::Pose3> {
+ private:
+  using This = ContactPoseFactor;
+  using Base = gtsam::NoiseModelFactor2<gtsam::Pose3, gtsam::Pose3>;
+
+  // The contact point reference frame in the link's CoM frame.
+  gtsam::Pose3 comTcontact_;
+
+ public:
+  /**
+   * Constructor.
+   *
+   * @param link_pose_key Key for the CoM pose of the link in contact.
+   * @param point_key Key for the contact point in the environment.
```

Comment:
I think a better name here is contact_pose_key, and the description should be "key for the contact point pose in the world/spatial frame"

Reply:
Awesome, I was wondering about that too. Glad you could confirm my doubts Dan 😊